sov-modules/sov-modules-api/src/default_signature.rs

```diff
@@ -1,50 +1,100 @@
 use crate::{SigVerificationError, Signature};
 use borsh::{BorshDeserialize, BorshSerialize};
+use ed25519_dalek::{
+    ed25519::signature::Signature as DalekSignatureTrait, PublicKey as DalekPublicKey,
+    Signature as DalekSignature,
+};
 
-// TODO: https://github.com/Sovereign-Labs/sovereign/issues/253
-#[derive(PartialEq, Eq, Clone, BorshDeserialize, BorshSerialize, Debug)]
-pub struct DefaultPublicKey {
-    pub(crate) pub_key: Vec<u8>,
-}
+use ed25519_dalek::{PUBLIC_KEY_LENGTH, SIGNATURE_LENGTH};
+
+#[cfg(feature = "native")]
+pub mod private_key {
+    use super::{DefaultPublicKey, DefaultSignature};
+    use ed25519_dalek::{Keypair, Signer};
+    use rand::rngs::OsRng;
 
-impl DefaultPublicKey {
-    pub fn new(pub_key: Vec<u8>) -> Self {
-        Self { pub_key }
+    pub struct DefaultPrivateKey {
+        key_pair: Keypair,
     }
 
-    pub fn sign(&self, _msg: [u8; 32]) -> DefaultSignature {
-        DefaultSignature {
-            msg_sig: vec![],
-            should_fail: false,
+    impl DefaultPrivateKey {
+        pub fn generate() -> Self {
+            let mut csprng = OsRng;
+
+            Self {
+                key_pair: Keypair::generate(&mut csprng),
+            }
+        }
+
+        pub fn sign(&self, msg: [u8; 32]) -> DefaultSignature {
+            DefaultSignature {
+                msg_sig: self.key_pair.sign(&msg),
+            }
+        }
+
+        pub fn pub_key(&self) -> DefaultPublicKey {
+            DefaultPublicKey {
+                pub_key: self.key_pair.public,
+            }
         }
     }
 }
 
-impl<T: AsRef<str>> From<T> for DefaultPublicKey {
-    fn from(key: T) -> Self {
-        let key = key.as_ref().as_bytes().to_vec();
-        Self { pub_key: key }
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct DefaultPublicKey {
+    pub(crate) pub_key: DalekPublicKey,
+}
+
+impl BorshDeserialize for DefaultPublicKey {
+    fn deserialize_reader<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let mut buffer = [0; PUBLIC_KEY_LENGTH];
+        reader.read_exact(&mut buffer).unwrap();
+
+        Ok(Self {
+            pub_key: DalekPublicKey::from_bytes(&buffer).unwrap(),
+        })
     }
 }
 
-#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, PartialEq, Eq, Debug, Clone, Default)]
+impl BorshSerialize for DefaultPublicKey {
+    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        writer.write_all(self.pub_key.as_bytes())
+    }
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
 pub struct DefaultSignature {
-    pub msg_sig: Vec<u8>,
-    pub should_fail: bool,
+    pub msg_sig: DalekSignature,
+}
+
+impl BorshDeserialize for DefaultSignature {
+    fn deserialize_reader<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let mut buffer = [0; SIGNATURE_LENGTH];
+        reader.read_exact(&mut buffer).unwrap();
+
+        Ok(Self {
+            msg_sig: DalekSignature::from_bytes(&buffer).unwrap(),
```

Comment:
Same here about panic:

```suggestion
            msg_sig: DalekSignature::from_bytes(&buffer)?,
```